pkg/dataobj/sections/indexpointers/row_reader.go

```diff
@@ -175,34 +175,26 @@ func translateIndexPointersPredicate(p RowPredicate, columns []dataset.Column) d
 	}
 
 	switch p := p.(type) {
-	case TimeRangePredicate:
+	case TimeRangeRowPredicate:
 		return convertTimeRangePredicate(p, minTimestampColumn, maxTimestampColumn)
 
 	default:
 		panic(fmt.Sprintf("unsupported predicate type %T", p))
 	}
 }
 
-func convertTimeRangePredicate(p TimeRangePredicate, minTimestampColumn, maxTimestampColumn dataset.Column) dataset.Predicate {
+func convertTimeRangePredicate(p TimeRangeRowPredicate, minTimestampColumn, maxTimestampColumn dataset.Column) dataset.Predicate {
 	return dataset.AndPredicate{
-		Left: dataset.OrPredicate{
-			Left: dataset.EqualPredicate{
+		Left: dataset.NotPredicate{
+			Inner: dataset.LessThanPredicate{
 				Column: minTimestampColumn,
-				Value:  dataset.Int64Value(p.MinTimestamp.Unix()),
-			},
-			Right: dataset.GreaterThanPredicate{
-				Column: minTimestampColumn,
-				Value:  dataset.Int64Value(p.MinTimestamp.Unix()),
+				Value:  dataset.Int64Value(p.Start.UnixNano()),
```

Comment:
Was this a bug that got fixed? Nice 🎉 I think we previously lost precision by using Unix() and now using UnixNano